validator/main.py

```diff
@@ -115,43 +120,48 @@ def get_args(self) -> Dict[str, Any]:
             "model_threshold": self._kwargs.get("model_threshold", 0.5),
         }
 
-    def get_topics_ensemble(self, text: str, candidate_topics: List[str]) -> List[str]:
+    def get_topics_ensemble(self, text: str, candidate_topics: List[str]):
         applicable_topics = []
 
-        for candidate_topic in candidate_topics:
-            candidates = [candidate_topic, "other"]
-            topic, confidence = self.get_topic_zero_shot(text, candidates)
-
-            if confidence < self._model_threshold:
+        topics, scores = self.get_topic_zero_shot(text, candidate_topics)
+        for topic, score in zip(topics, scores):
+            if score > self._model_threshold:
+                applicable_topics.append(topic)
+            elif score < self._model_threshold:
                 response = self.call_llm(text, candidate_topics)
                 topic = json.loads(response)["topic"]
-
-            if topic != "other":
+            if topic in self._invalid_topics:
                 applicable_topics.append(topic)
-
         return applicable_topics
 
-    def get_topics_llm(self, text: str, candidate_topics: List[str]) -> List[str]:
+    async def get_topics_llm(self, text: str, candidate_topics: List[str]) -> List[str]:
```

Comment:
This looks good. Before merging, can you benchmark the latency of this method vs. doing a single LLM call that returns multiple topics?

Will sync offline on benchmarking code.